types/category.ts

```diff
@@ -0,0 +1,6 @@
+export class Category {
+    category: string
+    constructor({ category } : { category: string }) {
+        this.category = category
+    }
+}
```

Comment:
Why are we creating classes if we dont want to do anything other than store data? Am I missing some context here?

Reply:
We can just do `const newCategory: Category = {...}` right?

Reply:
I am reasonably new to TS, so I probably incorrectly created the first Resource class, because I wanted to define the shape of the Resource that was created and passed around and that is what I am used to doing in JS. This could have just been a `type` or `interface` definition(?)
My guess is that @nheingit followed this pattern for the other items because of what I originally did, of course I could be wrong.

Reply:
@AidenMontgomery that's correct! 

I was going to suggest some changes but I think I'm not understanding what the intent is with some of these things. For example, will `Blockchain` ever have more data attached to it than `blockchain` as a `string`?

Reply:
Yeah, looking back I don't know why I did a class. This would 100% be an interface or type. I just was following the previously established pattern. 

I created all of the types to at least have a field so I could keep working, but each of these interfaces will map 1-to-1 with their Airtable table.

Reply:
I should have put a TODO on each of these, it was just late and I wanted to get to a decent stopping point.

Reply:
> @with-heart RE: class data — I created all of the types to at least have a field so I could keep working, but each of these interfaces will map 1-to-1 with their Airtable table.

That makes perfect sense! I had actually just seen a comment you made in Discord that made this click for me before I saw this comment 😆

Reply:
In that case, here's what I'd recommend:

1. Reduce them down to a single `types.ts` file. Not enough going on here to justify having a single file per type imo
2. Convert all classes to interfaces
3. Rename the keys to be a little more specific to the type of data (e.g. an `Author` doesn't have an `author`, it has a `name` or an `id` or something)

So essentially:

```ts
export interface Author {
  // `author` -> `name`
  // maybe `id`? not sure what it's meant to store
  name: string
}

export interface Blockchain {
  // `blockchain` -> `id`
  // maybe this should be `name`? not sure what the exact intent is
  id: string
}

export interface Category {
  // `category` -> `name`
  // maybe `id`? not sure of intent
  name: string
}

export interface Contributor {
  discordId: string
  devDaoId?: string
  ethAddress?: string
  solAddress?: string
  // maybe should be required and always initialize with an empty `[]`?
  resources?: Resource[]
  // same as `resources`?
  glossaries?: Glossary[]
  // does a contributor really have an `author`?
  author?: Author
  // same as `resources`?
  blockchains?: BlockChain[]
  // does a contributor really have `categories`?
  categories?: Category[]
  // does a contributor really have `tags`?
  tags?: Tag[]
}

export interface Glossary {
  // `glossary` -> `name`
  // maybe `id` instead?
  name: string
}

export interface Resource {
  contributor: Contributor
  author: Author
  title: string
  summary: string
  source: string
  level: string
  mediaType: string
  blockchain: Blockchain[]
  category: Category[]
  tags: Tag[]
}

export interface Tag {
  // `tag` -> `name`
  name: string
}
```

Reply:
This looks really good to me.
To better understand the props of each of the types, blockchain, category, etc, look at the forms for creating each of these that are linked from [this google doc](https://docs.google.com/document/d/1GwvpUAu-aFJSMW17sfbG6KV_QYQhVTeI2FkC1SoVXoo/edit)

Reply:
@AidenMontgomery Everything makes quite a bit more sense now lol, thanks for the link